### ISSUE TITLE
use custom HGPS sensor for GPS home

### ISSUE THREAD

**Issue by yaapu** [yaapu/HorusMappingWidget#11]
If a a HGPS (home GPS) sensor is found use it to draw a home marker on the map

**Comment by RichCattell** [comment]
Has this functionality been implemented?

Would this allow the home marker to be shown for the transmitter's position on the map?
This would be very useful to assist with finding a crashed model - I assume that a gps module would need to be added to the TX and connected to a spare UART - where should this be connected on a radiomaster TX16S?

**Comment by chucky-whiskers** [comment]
@RichCattell The TX16S has two configurable UART's under the "charge port" cover, as well as a UART in the external module bay, as well as (at least) two free UARTS internally. The best location to install a GPS is going to depend on a few factors. There are several videos on this topic available on YouTube, so I will not go too far in-depth, but any UART will work. if you do not wish to deal with trying to find a permanent internal location for a GPS, it may make more sense to just plug in to one of the UARTs on the bottom of the transmitter as needed. You just configure it like any other sensor in the Radio Settings menu.

**Comment by SilentZen** [comment]
Yes could an option to add getTxGPS() as a home marker be added, thanks.

**Comment by kasatka60** [comment]
I don't know how to program, but I changed some of the code.

```
function task4HzUpdateHome(widget, now)
  local GPSTable = getTxGPS()
  if GPSTable == nil then
    if telemetry.lat == nil then
      return
    end
    if telemetry.lon == nil then
      return
    end
    if status.homeSet then 
      return 
    end
    if telemetry.groundSpeed < conf.horSpeedMultiplier*0.5 then
      if not status.homeTimerRunning then
        status.homeTimerStart = getTime()
        status.homeTimerRunning = true
      else
        local elapsed = getTime() - status.homeTimerStart
      
        if elapsed >= 500 then
          telemetry.homeLat = telemetry.lat
          telemetry.homeLon = telemetry.lon
          status.homeSet = true
          status.homeTimerRunning = false
        end
      end
    else
      if status.homeTimerRunning  then
        status.homeTimerRunning = false
        status.homeTimerStart = 0
      end
    end
  else
    if GPSTable.numsat > 5 then
      	telemetry.homeLat = GPSTable.lat
		telemetry.homeLon = GPSTable.lon
		status.homeSet = true
	end
  end
end
```

We need to check what happened.
I honestly stole some code from TxGPStest

https://youtu.be/8Xkm52bt7mU

[OTX_ETX.zip](https://github.com/user-attachments/files/28315159/OTX_ETX.zip)
